pyinstrument/renderers/console.py

```diff
@@ -137,11 +147,64 @@ def render_frame(self, frame: Frame, indent: str = "", child_indent: str = "") -
 
         return result
 
-    def frame_proportion_of_total_time(self, frame: Frame):
-        return frame.time / self.root_frame.time
+    def render_frame_flat(self, frame: Frame) -> str:
+        @dataclass(frozen=True, eq=True)
+        class FrameDesc:
+            code_position_short: Optional[str]
+            function: str
```

Comment:
`frame.identifier` should do the job of uniquely identifying a frame's code location for the purpose of this computation, I don't think we'd need this dataclass.

Reply:
Thanks! I believe d52a191 addresses this.